dlt/pipeline/configuration.py

```diff
@@ -11,8 +11,8 @@
 class PipelineConfiguration(BaseConfiguration):
     pipeline_name: Optional[str] = None
     pipelines_dir: Optional[str] = None
-    destination_name: Optional[str] = None
-    staging_name: Optional[str] = None
+    destination_type: Optional[str] = None
```

Comment:
maybe we should just call this destination?

Reply:
I'd keep `_type` to be more explicit. also please link to this PR and add info that we changed the names in this config in here https://github.com/dlt-hub/dlt/issues/763

also please add `on_resolved` and please keep `destination_name` and `staging_name` in the config but display deprecation warnings. Steini added a method to warn on deprecation somewhere in the Pipeline class

Reply:
there are people that configure destinations via config.

btw. we may keep destination name in the longer run. so in fact we'll create a destination with given type and given name. so let's do that:
- if only name is present or only type is present, we create a destination with name == type
- if they are different we pass both as destination reference.
i think destination reference supports it

Reply:
Ok, so we do not deprecated destination_name and staging_name, right?